File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    int n = Integer.parseInt(br.readLine());
    StringTokenizer st = new StringTokenizer(br.readLine());
    int x = Integer.parseInt(br.readLine());

    int[] arr = new int[n];
    for(int i=0; i<n; i++){
      arr[i] = Integer.parseInt(st.nextToken());
    }
    Arrays.sort(arr);

    int count = 0;
    int frontIndex = 0;
    int behindIndex = n-1;
    while(frontIndex < behindIndex){
      int sum = arr[frontIndex] + arr[behindIndex];
      if(sum == x){
        count ++;
        behindIndex--;
      } else if (sum < x){
        frontIndex++;
      } else if (sum > x){
        behindIndex--;
      }
    }

    System.out.println(count);
  }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int arr[100002];
	int n; cin >> n;

	for(int i=0; i<n; i++) cin >> arr[i];
	sort(arr, arr+n);

	int num; cin >> num;

	int st = 0, end = n-1;

	int count = 0;
	while(st < end){
		if(arr[st] + arr[end] < num) st++;
		else if(arr[st] + arr[end] > num) end--;
		else if(arr[st] + arr[end] == num){
			count++;
			st++;
		}
	}

	cout << count;
}